src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -560,7 +560,10 @@ public void removeConsumerRecoveryListener(ConsumerRecoveryListener listener) {
     }
 
     private synchronized void beginAutomaticRecovery() throws InterruptedException {
-        this.wait(this.params.getRecoveryDelayHandler().getDelay(0));
+        final long delay = this.params.getRecoveryDelayHandler().getDelay(0);
+        if (delay > 0)  {
```

Comment:
i had tried to use a backoff recovery delay handler that had an initial delay of 0 and quickly realized that recovery would hang forever here. An easy work around is to use a 1ms initial delay, but this feels like a better solution that may save someone else in the future.

Reply:
This is reasonable if you ask me.